src/shared/utilities/timeoutUtils.ts

```diff
@@ -77,3 +77,28 @@ export class Timeout {
         }
     }
 }
+
+/**
+ * Invokes `fn()` until it returns a truthy value.
+ *
+ * @param fn  Function whose result is checked
+ * @param opt.timeout  Timeout in ms (default: 5000)
+ * @param opt.interval  Interval in ms between fn() checks (default: 500)
+ *
+ * @returns Result of `fn()`, or `undefined` if timeout was reached.
+ */
+export async function waitUntil<T>(
+    fn: () => Promise<T>,
+    opt: { timeout: number; interval: number } = { timeout: 5000, interval: 500 }
+): Promise<T | undefined> {
+    for (let i = 0; true; i++) {
+        const result: T = await fn()
+        if (result) {
+            return result
+        }
```

Comment:
The implementation used in this function benefits from the fact that a `false` return will fail this test and cause the function to loop, but what if we want to actively return a `false` value (or heck, even an `undefined`)? Should we enforce `if (result === undefined)` here and have the awaited function explicitly return `undefined`/should we return an object of `{value: T, doneWaiting: boolean}`?

Reply:
The caller has full control by how it defines `fn`.

Reply:
It doesn't, though. This function can't handle a `false` return since a `false` will cause this to keep looping. So say you have something like 
```
const doYouWantFriesWithThat = await waitUntil( async () => {
    if (doneLookingAtTheComboMenu) {
        return decision
    }
})
```

you'll never get a `false` response...you'll get `undefined` after going through the full interval, even though `false` is a valid return value.

Reply:
Nice catch. It seems like the more interesting thing is knowing when a gate is open, rather than some value in addition to the gate being open.

Reply:
Let's discuss in a followup PR. This PR is needed for integ tests on master.